### PR TITLE
Migrating to junit5

### PR DESCRIPTION
**Pull request by puthusseri**
I tried to upgraded the code base to junit5 to get new features for unit testing. Below are the steps, I followed :
**Steps:**

1. Update the dependencies in pom.xml
2. Changing Packages

> org.junit.Test to org.junit.jupiter.api.Test
> org.junit.Assert.* to org.junit.jupiter.api.Assertions.*
3. When converting exception assertions from JUnit 4 to JUnit 5, the approach changes from using the @Test(expected = ...) annotation to using the assertThrows method in JUnit 5

Screenshot 2024-11-01 at 1 15 06 AM

### CODE REVIEW

**Comment by puthusseri** [comment]
#231

**Comment by davidmoten** [comment]
Looks worthwhile, thanks! I'll review soon.

File: geo/src/test/java/com/github/davidmoten/geo/TestingUtil.java [hunk 1]
```diff
@@ -17,31 +16,14 @@ public class TestingUtil {
...
-            constructor.newInstance();
-        } catch (InstantiationException e) {
-            throw new RuntimeException(e);
-        } catch (IllegalAccessException e) {
-            throw new RuntimeException(e);
-        } catch (IllegalArgumentException e) {
-            throw new RuntimeException(e);
-        } catch (InvocationTargetException e) {
-            throw new RuntimeException(e);
-        }
```
Comment (puthusseri) [inline geo/src/test/java/com/github/davidmoten/geo/TestingUtil.java:44]: I missed this case.
Can I make change to this method?

Comment (davidmoten) [reply geo/src/test/java/com/github/davidmoten/geo/TestingUtil.java:44]: Sure

Comment (puthusseri) [reply geo/src/test/java/com/github/davidmoten/geo/TestingUtil.java]: @davidmoten That one missed change was also done.  You can proceed with the review.

**Review by davidmoten — APPROVED** [review]
Looks good, thanks!

**Comment by davidmoten** [comment]
oops, forgot to squash, I'll squash now

**Comment by davidmoten** [comment]
squashed, thanks for the contribution @puthusseri!

**Pull request merged**